src/DataBlobs/services/BlobStores.jl

```diff
@@ -94,21 +98,46 @@ function getBlob(dfg::AbstractDFG, entry::BlobEntry)
     end
     throw(
         KeyError(
-            "could not find $(entry.label), uuid $(entry.id)) in any of the listed blobstores:\n $([s->getKey(s) for (s,v) in stores]))"
+            "could not find $(entry.label), uuid $blobId) in any of the listed blobstores:\n $([s->getKey(s) for (s,v) in stores]))"
         )
     )
 end
 
-addBlob!(dfg::AbstractDFG, entry::BlobEntry, data::T) where T =
-        addBlob!(getBlobStore(dfg, entry.blobstore), entry, data)
+getBlob(store::AbstractBlobStore, entry::BlobEntry) =
+    getBlob(store, entry.originId)
+
+#add 
+addBlob!(dfg::AbstractDFG, entry::BlobEntry, data) =
+    addBlob!(getBlobStore(dfg, entry.blobstore), entry, data)
+
+addBlob!(store::AbstractBlobStore, entry::BlobEntry, data) =
+    addBlob!(store, entry.originId, data)
+    
+addBlob!(store::AbstractBlobStore, data) =
+    addBlob!(store, uuid4(), data)
+
+#fallback as not all blobStores use filename
+addBlob!(store::AbstractBlobStore, blobId::UUID, data, ::String) = 
```

Comment:
maybe just clarify blobId vs originId here

Reply:
this is just the id that is used in the blobstore as it can only have one. Dispatch on entry should sort out what to pass in here